packages/@expo/cli/src/verifyNativeAssets/verifyNativeAssets.ts

```diff
@@ -0,0 +1,172 @@
+import fs from 'fs';
+import glob from 'glob';
+import path from 'path';
+
+import {
+  BuildManifest,
+  ExportedMetadata,
+  ExportedMetadataAsset,
+  FullAssetDump,
+  FullAssetDumpEntry,
+  Platform,
+} from './types';
+import { CommandError } from '../utils/errors';
+
+const debug = require('debug')('expo:verify-native-assets') as typeof console.log;
+
+/**
+ * Finds any assets that will be missing from an app given a build and an exported update bundle.
+ *
+ * @param buildPath Path to an EAS or local build containing an expo-updates embedded manifest (app.manifest)
+ * @param exportPath Path to a directory produced by the command `npx expo export --dump-assetmap`
+ * @param platform Either `android` or `ios`
+ * @param projectRoot The project root path
+ * @returns An array containing any assets that are found in the Metro asset dump, but not included in either app.manifest or the exported bundle
+ */
+export function getMissingAssets(
+  buildPath: string,
+  exportPath: string,
+  platform: Platform,
+  projectRoot: string
+) {
+  const buildManifestHashSet = getBuildManifestHashSet(
+    getBuildManifest(buildPath, platform, projectRoot)
+  );
+
+  const fullAssetMap = getFullAssetDump(exportPath);
+  const fullAssetSet = getFullAssetDumpHashSet(getFullAssetDump(exportPath));
+
+  const exportedAssetSet = getExportedMetadataHashSet(getExportedMetadata(exportPath), platform);
+
+  debug(`Assets in build: ${JSON.stringify([...buildManifestHashSet], null, 2)}`);
+  debug(`Assets in exported bundle: ${JSON.stringify([...exportedAssetSet], null, 2)}`);
+  debug(`All assets resolved by Metro: ${JSON.stringify([...fullAssetSet], null, 2)}`);
+
+  const buildAssetsPlusExportedAssets = new Set(buildManifestHashSet);
+  exportedAssetSet.forEach((hash) => buildAssetsPlusExportedAssets.add(hash));
+
+  const missingAssets: FullAssetDumpEntry[] = [];
+
+  fullAssetSet.forEach((hash) => {
+    if (!buildAssetsPlusExportedAssets.has(hash)) {
+      const asset = fullAssetMap.get(hash);
+      console.warn(`  Missing asset: hash = ${hash}, file = ${asset?.files[0] ?? ''}`);
+      asset && missingAssets.push(asset);
+    }
+  });
+
+  return missingAssets;
+}
+
+/**
+ * Reads and returns the embedded manifest (app.manifest) for a build.
+ *
+ * @param buildPath Path to the build folder
+ * @param platform Either 'android' or 'ios'
+ * @param projectRoot The project root path
+ * @returns the JSON structure of the manifest
+ */
+export function getBuildManifest(buildPath: string, platform: Platform, projectRoot: string) {
+  let realBuildPath = buildPath;
+  if (buildPath === projectRoot) {
+    switch (platform) {
+      case 'android':
+        realBuildPath = path.resolve(projectRoot, 'android', 'app', 'build');
+        break;
+      default:
```

Comment:
Even though we type it to only `ios | android`, it might make sense to explicitly support `ios` instead of defaulting to it. That way, if ]platform is ever changed to something like `macos`]
(https://github.com/expo/expo/pull/26292), it is not silently inheriting `ios`.

Reply:
Platform is now a required parameter. I'll take a look to see if this code can be adjusted.